weave-js/src/components/ToggleButtonGroup.tsx

```diff
@@ -49,34 +53,39 @@ export const ToggleButtonGroup = React.forwardRef<
         onValueChange={handleValueChange}
         className="flex gap-px"
         ref={ref}>
-        {options.map(({value: optionValue, icon}) => (
-          <ToggleGroup.Item
-            key={optionValue}
-            value={optionValue}
-            disabled={isDisabled || value === optionValue}>
-            <Button
-              icon={icon}
-              size={size}
-              className={twMerge(
-                size === 'small' &&
-                  (icon ? 'gap-4 px-4 py-3 text-sm' : 'px-8 py-3 text-sm'),
-                size === 'medium' &&
-                  (icon ? 'gap-5 px-7 py-4 text-base' : 'px-10 py-4 text-base'),
-                size === 'large' &&
-                  (icon
-                    ? 'gap-6 px-10 py-8 text-base'
-                    : 'px-12 py-8 text-base'),
-                isDisabled && 'pointer-events-none opacity-35',
-                value === optionValue
-                  ? 'bg-teal-300/[0.48] text-teal-600 hover:bg-teal-300/[0.48]'
-                  : 'hover:bg-oblivion/7 bg-oblivion/5 text-moon-600 hover:text-moon-800',
-                'first:rounded-l-sm', // First button rounded left
-                'last:rounded-r-sm' // Last button rounded right
-              )}>
-              {optionValue}
-            </Button>
-          </ToggleGroup.Item>
-        ))}
+        {options.map(
+          ({value: optionValue, icon, isDisabled: optionIsDisabled}) => (
+            <ToggleGroup.Item
+              key={optionValue}
+              value={optionValue}
+              disabled={isDisabled}
+              asChild>
```

Comment:
asChild is needed to prevent a `validateDomNesting` error for nested buttons, see https://www.radix-ui.com/primitives/docs/guides/composition